File: AppleEsportsErp/src/AppleEsportsErp.Application/Constants/MemberWalletRules.cs
namespace AppleEsportsErp.Application.Constants;

/// <summary>
/// Thresholds that govern a member's Gaming-wallet-funded session. The customer overlay
/// mirrors these in client/src/utils/memberWalletRules.js — change both together.
/// </summary>
public static class MemberWalletRules
{
    /// <summary>
    /// A member session cannot be started below this Gaming balance. Matches the auto-stop
    /// floor, so a session can never start only to be stopped on its first tick.
    /// </summary>
    public const decimal MinimumGamingBalanceToStart = 1m;

    /// <summary>Remaining Gaming balance at which the session is stopped automatically.</summary>
    public const decimal AutoStopRemainingBalance = 1m;

    /// <summary>First low-balance reminder — also triggers the member email + operator alert.</summary>
    public const decimal FirstReminderRemainingBalance = 20m;

    /// <summary>Second (final) low-balance reminder to the member.</summary>
    public const decimal SecondReminderRemainingBalance = 10m;
}

File: client/src/utils/memberWalletRules.js
// Mirrors AppleEsportsErp.Application.Constants.MemberWalletRules (backend) so the overlay
// stops/warns at exactly the same balances the server refuses to start a session at.
// Change both together.

// A member session cannot start below this Gaming balance — and is auto-stopped once the
// remaining balance falls under it, so a session can never start only to stop immediately.
export const MIN_GAMING_BALANCE_TO_START = 1;
export const AUTO_STOP_REMAINING_BALANCE = 1;

// Two reminders, in the order they fire. The first also triggers the member email +
// operator alert on the backend.
export const FIRST_REMINDER_REMAINING_BALANCE = 20;
export const SECOND_REMINDER_REMAINING_BALANCE = 10;

// Minutes of play a remaining balance still buys at the PC's hourly rate.
export function minutesRemainingFor(remainingBalance, ratePerHour) {
  const rate = Number(ratePerHour) || 0;
  const remaining = Number(remainingBalance) || 0;
  if (rate <= 0 || remaining <= 0) return 0;
  return Math.max(0, Math.floor((remaining / rate) * 60));
}
